spec/integration/lib/auth0/auth0_client_spec.rb

```diff
@@ -42,4 +42,28 @@
     let(:credentials) { v2_credentials.merge({access_token: ENV["MASTER_JWT"]}) }
   end
 
+  context "client headers" do
+    let(:client) { Auth0::Client.new(v2_credentials.merge({access_token: 'abc123', domain: 'myhost.auth0.com'})) }
+    let(:headers) { client.class.headers }
+
+    it "has the correct headers present" do
+      headers.keys.sort.should eql ['Authorization', 'Content-Type', 'User-Agent', 'X-Auth0-Client']
+    end
+
+    it "uses the correct access token" do
+      headers['Authorization'].should eql "Bearer abc123"
+    end
+
+    it "is always json" do
+      headers['Content-Type'] = 'application/json'
```

Comment:
Seriously funny :smile: 
… I think I did this when I was hanging with @eugeniop and @ntotten. I'll amend :-)